faunadb-common/src/main/java/com/faunadb/common/models/tags/Tag.java

```diff
@@ -0,0 +1,84 @@
+package com.faunadb.common.models.tags;
+
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Tag {
+
+    private static final int KEY_CHAR_LIMIT = 40;
+    private static final int VALUE_CHAR_LIMIT = 80;
+
+    private static final Pattern ALLOWABLE_CHARACTER_REGEX = Pattern.compile("^\\w+$");
+    private String key;
+    private String value;
+
+    public Tag() {
+        this.key = "";
+        this.value = "";
+    }
```

Comment:
I don't think we need this option. As an empty tag seems invalid.

Let's only vend constructors and factories that produce valid objects.

```suggestion
```